### PR TITLE
[mathbox] Various arithmetic theorems + revise elpwbi

### PR DESCRIPTION
**Pull request by icecream17**
Various arithmetic theorems spurred on by trying to multiply faster (or in this case of `1t1e1ALT`, investigating `ax-mulcom`).
In practice I've only found `sqn5ii` (and I guess `sqn5i`) to help. Naming was hard as always.

(Also just noticed that `elpwbi` is just `elpw2i` commuted.)

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -609582,6 +609582,66 @@ fixed reference functional determined by this vector (corresponding to
...
+      KJZDDKJZIJLADKJZKJZIJZBBKJACKJZUKIJZADEFMUIBUIBKGGUAUNUJUMIJZUKIJUPUJUKUM
+      AAEENDDFFNLULOADEFNNUBUQUOUKIUOAADDIJZIJZKJUJAURKJZIJUQCUSAKBDIJUIDIJCUSB
+      UIDIUIBGPQHADDEFFUCRSAAUREEDDFFUDUEUTUMUJIUTALDKJZKJUMURVAAKVAURDFUFPSALD
+      EOFUGTSUHQTR $.
+  $}
+
+  ${
+    sqn5i.1 $e |- A e. NN0 $.
+    $( The square of a number ending in 5.  This shortcut only works because 5
+       is half of 10. (Contributed by Steven Nguyen, 16-Sep-2022.) $)
```
Comment (icecream17) [inline set.mm:609625]:
```suggestion
       is half of 10.  (Contributed by Steven Nguyen, 16-Sep-2022.) $)
```
(easier to commit a review than to go back and rewrap and commit)

File: set.mm [hunk 2]
```diff
@@ -609582,6 +609582,66 @@ fixed reference functional determined by this vector (corresponding to
...
+      ( c5 cdc cmul co cc0 caddc c2 0nn0 deccl nn0cni 5nn0 eqid addid2i decaddi
+      5cn 2nn0 cn0 eqtri c1 5p5e10 decaddci2 diffsqi 5t5e25 peano2nn0 nn0mulcli
+      wcel ax-mp decmulnc mul01i deceq2i 2cn mul02i oveq1i decma ) ACDZUQEFAGDZ
+      AUAHFZGDZEFCCEFZHFAUSEFZIDZCDURUQUTCURAGBJKLQAGCURCBJMURNZCQOZPACUSUQCBMM
+      UQNUSNUBUCUDAGICUTVCCURVABJRMVDUEUSGASUHUSSUHBAUFUIZJKZVBGIAUTEFZIAUSBVFU
+      GJRVHVBAGEFZDVBGDUSGABVFJUJVIGVBAABLUKULTIUMOPGUTEFZCHFGCHFCVJGCHUTUTVGLU
+      NUOVETUPT $.
+
+    sqn5ii.2 $e |- ( A + 1 ) = B $.
+    sqn5ii.3 $e |- ( A x. B ) = C $.
```
Comment (benjub) [inline set.mm:609636]: Your mathbox your choice, but I would suggest replacing these two hypothses with
```
$e |- B = ( ( A x. ( A + 1 ) )  $.
```
(in local definitions, it's usual to put the definiendum on the LHS)

**Review by benjub — APPROVED** [review]

File: set.mm [hunk 3]
```diff
@@ -609582,6 +609582,66 @@ fixed reference functional determined by this vector (corresponding to
...
+      AGFBBGFEFZABGFZQEFZEFPHQGFZEFABABCDCDIRSPESRQABCDJKLMN $.
+  $}
+
+  ${
+    diffsqi.a $e |- A e. CC $.
+    diffsqi.n $e |- N e. CC $.
+    diffsqi.b $e |- ( A + N ) = B $.
+    diffsqi.c $e |- ( B + N ) = C $.
+    $( Difference of squares.  (Contributed by Steven Nguyen, 20-Sep-2022.) $)
+    diffsqi $p |- ( B x. B ) = ( ( A x. C ) + ( N x. N ) ) $=
```
Comment (avekens) [inline set.mm:609613]: Where is the "difference"? ( ( B x. B ) - ( N x. N ) ) = ( A x. C )?

B^2 = A^2 + 2AN + N^2 
=> B^2 - N^2 = A^2 + 2AN = A ( A + 2N). Why not choosing C = ( A + ( 2 x. N ) ) directly? See also comment of @benjub about ~sqn5ii

Comment (icecream17) [reply set.mm:609613]: Really I created the theorem first and then the name/comments. I'm not sure how to describe this theorem

File: set.mm (hunk 2 above)
Comment (icecream17) [reply set.mm:609636]: The applications I had in mind were multiplying numbers ending in 5. Would the current statements be slower than the combined hypothesis?

File: set.mm (hunk 3 above)
Comment (benjub) [reply set.mm:609613]: "Value of the square of the midterm of a 3-term arithmetic progression." ?

**Review by benjub — APPROVED** [review]

**Pull request merged**